### PR TITLE
Minor update to Support spells scripts

### PR DESCRIPTION
**Pull request by raymondtfr**
* Updated ``find_person.lua`` to use Lua ternary operation in one case and also replaced whitespaces with tabs.
* Updated ``food.lua``.
* Replaced ``Variant.getPosition(variant)`` with ``variant:getPosition()``

### CODE REVIEW

File: data/spells/scripts/support/animate_dead_rune.lua [hunk 1]
```diff
@@ -18,6 +18,6 @@ function onCastSpell(creature, variant, isHotkey)
 	end
 
 	creature:getPosition():sendMagicEffect(CONST_ME_POFF)
-	creature:sendCancelMessage(RETURNVALUE_NOTPOSSIBLE)
+	creature:sendTextMessage(MESSAGE_STATUS_SMALL, Game.getReturnMessage(RETURNVALUE_NOTPOSSIBLE))
```
Comment (ninjalulz) [inline data/spells/scripts/support/animate_dead_rune.lua]: Is there any specific reason to this change?

Comment (raymondtfr) [inline data/spells/scripts/support/animate_dead_rune.lua]: To not use any old function from compat, and instead use the original function. (sendCancelMessage uses the sendTextMessage inside it, so its better to use the original function as the old one may be deprecated sometime?)

Comment (ninjalulz) [inline data/spells/scripts/support/animate_dead_rune.lua]: [Player.sendCancelMessage](https://github.com/otland/forgottenserver/blob/master/data/lib/core/player.lua#L58-L63) is not a compatibility function though, and it's widely used in the data pack & source code.

Comment (ranisalt) [inline data/spells/scripts/support/animate_dead_rune.lua]: It's not from compat, `sendCancelMessage` is different because it gets the message from a predefined list.

File: data/spells/scripts/support/food.lua [hunk 2]
```diff
@@ -1,19 +1,19 @@
-local food = {
+local foods = {
 	2666, -- meat
 	2671, -- ham
 	2681, -- grape
 	2674, -- apple
 	2689, -- bread
 	2690, -- roll
-	2696  -- cheese
+	2696 -- cheese
```
Comment (ranisalt) [inline data/spells/scripts/support/food.lua]: You should add a trailing comma instead to keep it consistent and aligned.

File: data/spells/scripts/support/food.lua [hunk 3]
```diff
@@ -1,19 +1,19 @@
...
 }
 
 function onCastSpell(creature, variant)
-	creature:addItem(food[math.random(#food)])
-	if math.random(1, 100) > 50 then
-		creature:addItem(food[math.random(#food)])
+	if math.random(1, 100) >= 50 then
```
Comment (ranisalt) [inline data/spells/scripts/support/food.lua]: You changed the chance from 50% to 51% by replacing `>` with `>=`. Also, as the chance is 50%, you can use `math.random(1, 2) == 1` instead.

Why did you change the order, exactly?

Comment (raymondtfr) [inline data/spells/scripts/support/food.lua]: Not exactly, ``math.random(1, 100)`` is a closed interval [a, b], so that [1, 100]  (has a ``b - 1 + 1`` which is ``100 - 1 + 1`` that is 100 different numbers (including 100), so ``>=`` could be read in a reversed order: ``50 <= math.random(1, 100)``, so if the random value was **50** we would get ``50 <= 50`` (b - a + 1) which is 1/2.
If I used ``<`` it would be a open interval (b - a), so it would exclude one element.

I had done it wrong when I updated this script a while ago, so I have now corrected the change.
The code is clearer with that statement moved below the ``if`` I guess.

Comment (ranisalt) [inline data/spells/scripts/support/food.lua]: Oh yeah, I'm more used to working with 0-based indexes and ranges and Lua patterns still confuse me :P

File: data/spells/scripts/support/find_person.lua [hunk 4]
```diff
@@ -79,10 +79,8 @@ function onCastSpell(creature, variant)
 		elseif absValue < 2.4142 then
 			if distanceValue > 0 then
 				direction = offset.y > 0 and DIRECTION_NORTHWEST or DIRECTION_SOUTHEAST
-			elseif offset.x > 0 then
-				direction = DIRECTION_SOUTHWEST
 			else
-				direction = DIRECTION_NORTHEAST
+				direction = offset.x > 0 and DIRECTION_SOUTHWEST or DIRECTION_NORTHEAST
```
Comment (ranisalt) [inline data/spells/scripts/support/find_person.lua]: Though I usually complain about ternaries, this is a particularly nice usage 😋

File: data/spells/scripts/support/animate_dead_rune.lua [hunk 5]
```diff
@@ -1,5 +1,5 @@
 function onCastSpell(creature, variant, isHotkey)
```
Comment (ranisalt) [inline data/spells/scripts/support/animate_dead_rune.lua]: While you're at it, `isHotkey` is not needed here too.

File: data/spells/scripts/support/destroy_field_rune.lua [hunk 6]
```diff
@@ -1,8 +1,7 @@
 function onCastSpell(creature, variant, isHotkey)
```
Comment (ranisalt) [inline data/spells/scripts/support/destroy_field_rune.lua]: Also here.

File: data/spells/scripts/support/animate_dead_rune.lua (hunk 5 above)
Comment (raymondtfr) [inline data/spells/scripts/support/animate_dead_rune.lua]: But this is a rune spell #1020 , it was originally made for rune spells, so I just removed those from instant spells.

Comment (ranisalt) [inline data/spells/scripts/support/animate_dead_rune.lua]: Oh, sure, didn't noticed it's a rune.

**Comment by ranisalt** [comment]
Thanks!
